File: 参考/cook/app/Services/FavoriteService.php
<?php
namespace App\Services;

use App\Repositories\FavoriteRepository;

class FavoriteService
{
    private $repo;

    public function __construct()
    {
        $this->repo = new FavoriteRepository();
    }

    public function getList($userId, $page = 1, $pageSize = 30)
    {
        $page = max(1, intval($page));
        $pageSize = max(1, intval($pageSize));

        $offset = ($page - 1) * $pageSize;
        $list = $this->repo->getByUser($userId, $offset, $pageSize);
        $total = $this->repo->countByUser($userId);

        return [
            'list' => $list,
            'total' => $total
        ];
    }

    public function add($userId, $recipeId)
    {
        if (!$recipeId) {
            throw new \Exception("recipe_id 不能为空");
        }

        $this->repo->insertIgnore($userId, $recipeId);
    }

    public function remove($userId, $recipeId)
    {
        $this->repo->delete($userId, $recipeId);
    }
}

File: backend/app/services/favorite_service.py
"""收藏业务逻辑层"""
from typing import Optional

from app.repositories.favorite_repository import FavoriteRepository
from app.schemas.favorite import FavoriteResponse, FavoriteListResponse


class FavoriteService:
    """收藏服务类"""

    def __init__(self, repository: FavoriteRepository):
        self.repository = repository

    def list_favorites(self, household_id: str, page: int = 1, page_size: int = 30) -> FavoriteListResponse:
        """收藏列表"""
        offset = (page - 1) * page_size
        items = self.repository.get_by_household(household_id, offset, page_size)
        total = self.repository.count_by_household(household_id)
        return FavoriteListResponse(
            data=[FavoriteResponse(**item) for item in items],
            total=total, page=page, page_size=page_size
        )

    def add_favorite(self, household_id: str, recipe_id: str) -> Optional[FavoriteResponse]:
        """收藏（幂等）"""
        favorite = self.repository.add(household_id, recipe_id)
        if not favorite:
            return None
        return FavoriteResponse(
            id=favorite.id,
            recipe_id=favorite.recipe_id,
            recipe_title="",
            cover=None,
            created_at=favorite.created_at,
        )

    def remove_favorite(self, household_id: str, recipe_id: str) -> bool:
        """取消收藏"""
        return self.repository.delete(household_id, recipe_id)
